### DISCUSSION TITLE
Testing infrastructure for installer

### DISCUSSION THREAD

**Discussion by kisst** [modoboa/modoboa#2431] [Installer]
First time users, and in general sys admins have low tolerance for first try error, and the current stability of the installer isn't the A+ ( hopefully not saying anything new with that ) I understand that containers are not suited for testing due to the docker vs systemd pid 0 madness, hence testing infra would need to be VM based which might be outside of the project budget however I think there would be value in it and would be happy to help with setting up the orchestration and testing capability if the project do consider the installer end2end testing, and clearly unit testing the installer as such just can't cover it to the level needed.
So is there an appetite for installer automated testing ?

**Comment by tonioo** [comment]
@kisst Yes, definitely! Is that something you could provide?

**Comment by kisst** [comment]
I can cover the IaC for it for sure, in terms of build capacity I think I might able cover arm builds, however I guess I will start with getting the CI/CD and IaC set up first, I presume the GitHub Actions is the preferred for this right ?

**Reply by kisst** [reply]
Just seen that Travis is used for it now, do you prefer to keep that or ... GH Actions ?

**Reply by tonioo** [reply]
We're (slowly) moving from travis to GH actions because they changed their pricing policy. However, using GH actions also for the e2e tests will consume more credits and we are on a free plan...

**Reply by kisst** [reply]
What's the preferred execution environment for this project, I would love to help out and bring some consistency into this project.

**Reply by tonioo** [reply]
Debian based distributions.

**Reply by kisst** [reply]
My bad not precise enough question, not asking about the application running env, but the CI/CD pipeline location. btw I'm on Debian too, and I guess Ubuntu would be included too, but should I try to make it work in GH Actions, or Travis, or Jenkins, or Drone.io or should I just make a local bash version with Virtualbox, or Vagrant or similar ?
